### PR TITLE
Add troubleshooting step for invalid request

### PR DESCRIPTION
**Pull request by dshokouhi**
## Proposed change

I went through the install steps today.  For the Authorized Redirect URI I used the nabu casa callback URL.  When I got to the final step I was actually accessing the UI via IP and ended up getting `Error 400: Invalid Request` along with the callback URL showing as my internal IP vs the nabu casa URL that I used.  

Referring back to the original step by step post it seems that this was what I was missing: https://maartendamen.com/ha-nest-sdm/

Sure enough logging into the HA frontend and using the nabu casa remote URL allowed me to setup the integration as expected.

I know that it was mentioned local IP should work but seems Google requires an actual domain name for the oauth redirect URI.  Maybe this was changed recently?

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by dshokouhi** [comment]
Umm is lint saying I misspelled `error` ? lol

File: source/_integrations/nest.markdown [hunk 1]
```diff
@@ -202,6 +202,8 @@ Once your developer account is set up and you have a valid `nest` entry in `conf
 
 - *Subscriber error: Subscription misconfigured. Expected topic name to match ...* means that the topic name in the Google Cloud Console was entered incorrectly. The topic name comes from the Device Console and must start with `projects/sdm-prod/topics/`. It is easy to make the mistake of creating a new topic rather than manually entering the right topic name.
 
+- *Error 400: Invalid Request* This error typically happens when there is a mismatch with how you are accessing Home Assistant and the Authorized Redirect URI you entered for OAUTH. If you used a publicly facing URL make sure that you are accessing Home Assistant with the same URL when you setup the Integration. You will definitely hit this error if you are accessing Home Assistant by local IP during intial setup.
```
Comment (allenporter) [inline source/_integrations/nest.markdown]: Since this is URL related, perhaps this should go up next to the Error 500 above?

**Comment by allenporter** [comment]
I think there was some confusion -- this has always required a public domain name, but technically the oauth spec doesn't need this, which is why there was some confusion.  Its something google requires on top.

**Comment by dshokouhi** [comment]
Ok that makes sense, it just really threw me off when I saw the 400 error so I ended up double checking the entire setup lol.  Not sure if tis worth mentioning this in an earlier step but since nobody else opened an issue it could just be my own mistake lol.

File: source/_integrations/nest.markdown (hunk 1 above)
Comment (dshokouhi) [reply source/_integrations/nest.markdown]: Done, thank you :)

**Review by allenporter — DISMISSED** [review]

File: source/_integrations/nest.markdown [hunk 2]
```diff
@@ -185,6 +185,8 @@ Once your developer account is set up and you have a valid `nest` entry in `conf
 
 - For general trouble with the SDM API OAuth authorization flow with Google, see [Troubleshooting](https://developers.google.com/nest/device-access/authorize#troubleshooting).
 
+- *Error 400: Invalid Request* This error typically happens when there is a mismatch with how you are accessing Home Assistant and the Authorized Redirect URI you entered for OAUTH. If you used a publicly facing URL make sure that you are accessing Home Assistant with the same URL when you setup the Integration. You will definitely hit this error if you are accessing Home Assistant by local IP during initial setup.
```
Comment (frenck) [inline source/_integrations/nest.markdown]:
```suggestion
- *Error 400: Invalid Request* This error typically happens when there is a mismatch between how you are accessing Home Assistant and the Authorized Redirect URI you entered for OAuth. If you used a publicly facing URL, make sure that you are accessing Home Assistant with the same URL when you set up the integration. You will definitely hit this error if you are accessing Home Assistant by local IP during the initial setup.
```

**Review by frenck — APPROVED** [review]
Thanks, @dshokouhi 👍 

Happy Holidays 🎄

**Pull request merged**